### DISCUSSION TITLE
Formatting RaSCSI Volumes

### DISCUSSION THREAD

**Discussion by marciot** [akuker/RASCSI#742] [Q&A]
I am really struggling with creating RaSCSI volumes with the following features:

- Using "Apple HD SC Setup"
- Setting arbitrary sizes
- Compatible with Basillisk II for data transfer
- Preferably doable under System 7.0.1

I have had some success using APS Power Tools or Silver Lining to format arbitrarily sized drives, but since this is not an Apple tool, I am not sure about compatibility with things like the Apple IIe card, which is what I am trying to format for now.

I tried the "patched" versions of Apple HD SC Setup 7.5.3 (which does not run under 7.0.1), but this fails to recognize the RaSCSI drives which are created with "Create Empty Disk Image File". What does work is to use "Create Named Drive" and select "Miniscribe M8425", but this limits me to *exactly* 20MBs.

So, does anyone have any ideas? On a scale of 1 to 10, I would currently rate the difficulty of formatting drives under classic MacOS as an 11! 😛

**Comment by MrGasS** [comment]
If you want, I uploaded some Apple HD SC pre-formatted images here: https://github.com/MrGasS/MacintoshSCSIBlankHardfiles

**Reply by marciot** [reply]
@MrGasS Do you mind sharing the techniques you used to create these?

**Reply by MrGasS** [reply]
Yes, of course! I'll try to be as clear as possible.

First, I create the blank image with this command under Linux > dd if=/dev/zero of=filename.img bs=1M count=1024
This command triggers the DD utility to make a blank image, in this example I'm creating a 1 GB image where "count=1024" is the size in megabytes. You can also do this with the DD utility for Windows or also on a modern Macintosh computer.

After I made the image, I bring it into Windows and I mount it as a SCSI Hard Disk on pce-macplus emulator.
In order to format the disk, I mount the patched Apple HD SC Setup bootable diskette image.
You can also use the SoftMac emulator, it's really useful for images bigger than 2GB as the pce-macplus emulator struggles a bit with them.

Hope this helps :)

**Reply by marciot** [reply]
@MrGasS: I’ll give that a try! It’s odd that doing the exact same process by creating the image in RaSCSI and running thr patched utility on a real Mac fails. Maybe PCE is using a special drive identifier the Apple Utility recognizes. Maybe if I find out what that is, we can add it to RaSCSI so we can create images from the RaSCSI web utility.

**Reply by MrGasS** [reply]
I know, it's an odd thing, but in 5 years I never figured out how to fix this problem.

I know that we're talking about the RaSCSI, but the same problem occurs with the BlueSCSI, and it even occured few years ago when I tested the ArdSCSIno, the BlueSCSI's father.

In fact, I did my first pre-formatted images in 2017/18 with my SCSI2SD due to this problem.
As I remember, I also used them when I tested the original RaSCSI from Gimons back in 2017.
